pkg/virt-handler/vm.go

```diff
@@ -291,6 +292,12 @@ func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, shouldDeleteVm bool) erro
 		return err
 	}
 
+	// Map Container Registry Disks to block devices Libvirt can consume
+	vm, err = registrydisk.MapRegistryDisks(vm)
+	if err != nil {
+		return err
+	}
```

Comment:
> I'm don't seeing using REST as an item that blocks you - you could easily just implement a rest API in this patch series now. I would much rather see that generic approach taken straight away than to add this code that we know someone will have to rewrite later.

We are not certain that someone will have to re-write this later.  Likewise, we are not certain that introducing a REST api for this feature won't result in someone having to re-write the api part out later because it's found to be unnecessary.

The implementation, as it is now, is expressed naturally without the need for a REST api. If we see that REST api integration makes sense for this feature in the future, then we'll adapt.

> Trying the REST approach straightaway, will also demonstrate the viability of that generic approach to dynamic storage integration

I'd prefer not to attempt to prove the viability of a generic storage REST api *(that I'm not 100% sold on yet)* in a feature that doesn't need it.

I do however agree that the generic storage REST api is a good idea for us to continue discussing in a channel outside of this PR. I'd like for these to be treated as separate non conflicting discussions *(hence the disclaimer a the top of this PR about the #311 proposal)*